sdk/daml-script/test/daml/upgrades/Interfaces.daml

```diff
@@ -0,0 +1,237 @@
+import DA.Foldable
+import DA.Optional
+import DA.Text
+
+-- Fixed package containing only the interface
+{- PACKAGE
+name: fixed-interface
+versions: 1
+-}
+
+{- MODULE
+package: fixed-interface
+contents: |
+  module FixedInterface where
+
+  data IV = IV with
+    owner : Party
+    version : Text
+
+  interface I where
+    viewtype IV
+
+    getVersion : Text
+    nonconsuming choice GetVersion : Text
+      controller (view this).owner
+      do
+        pure $ getVersion this
+-}
+
+-- Another fixed package containing a helper template for exercising the interface in a choice
+{- PACKAGE
+name: fixed-interface-caller
+versions: 1
+depends: fixed-interface-1.0.0
+-}
+
+{- MODULE
+package: fixed-interface-caller
+contents: |
+  module FixedInterfaceCaller where
+
+  import FixedInterface
+
+  template Caller with
+      party : Party
+    where
+    signatory party
+
+    choice CallInterface : Text with
+        cid : ContractId I
+      controller party
+      do
+        exercise cid GetVersion
+-}
+
+-- The versioned/upgraded package
+{- PACKAGE
+name: interfaces
+versions: 2
+depends: fixed-interface-1.0.0
+-}
+
+{- MODULE
+package: interfaces
+contents: |
+  module Interfaces where
+
+  import FixedInterface
+  import DA.Optional
+
+  template FITemplate with
+      party : Party
+    where
+    signatory party
+
+    interface instance I for FITemplate where
+      view = IV party "V1" -- @V 1
+      view = IV party "V2" -- @V  2
+      getVersion = "V1"    -- @V 1
+      getVersion = "V2"    -- @V  2
+
+    choice FetchFromInterface : Text with
+        cid : ContractId I
+      controller party
+      do
+        (_, res) <- fromSomeNote "Failed to fetch contract" <$> fetchFromInterface @FITemplate cid
+        let v = view $ toInterface @I res
+        pure v.version
+-}
+
+-- Alt templates package to test multiple preference at once
+{- PACKAGE
+name: interfaces-alt
+versions: 2
+depends: fixed-interface-1.0.0
+-}
+
+{- MODULE
+package: interfaces-alt
+contents: |
+  module InterfacesAlt where
+
+  import FixedInterface
+
+  template FITemplateAlt with
+      party : Party
+    where
+    signatory party
+
+    interface instance I for FITemplateAlt where
+      view = IV party "V1" -- @V 1
+      view = IV party "V2" -- @V  2
+      getVersion = "V1"    -- @V 1
+      getVersion = "V2"    -- @V  2
+-}
+
+interfacesV1 : PackageId
+interfacesV1 = getPackageId "interfaces-1.0.0"
+
+interfacesAltV1 : PackageId
+interfacesAltV1 = getPackageId "interfaces-alt-1.0.0"
+
+interfacesV2 : PackageId
+interfacesV2 = getPackageId "interfaces-2.0.0"
+
+interfacesAltV2 : PackageId
+interfacesAltV2 = getPackageId "interfaces-alt-2.0.0"
+
+-- cant upgrade contracts in interface choice argument - i dont think this is possible in daml-script
+main : TestTree
+main = tests
+  [ brokenOnIDELedger ("Calling an interface choice uses the highest version implementation at command level", defaultHighestVersionChoiceImplementationCommand)
```

Comment:
Probably best to add "By default" to these test names, I was confused at first until I read the implementation names